### ISSUE TITLE
Getting a `PairingError: Error=BackOff` while trying to pair

### ISSUE THREAD

**Issue by courentin** [maxileith/homebridge-appletv-enhanced#129]
### Analysis

I haven't been able to pair my device because of the error described below.
If I understand correctly the device asks for a backoff time before retrying to pair but the plugin retries 1 min after

It seems there's some logic to handle this case, but I'm not sure it is working because the following warning does not show up:
https://github.com/maxileith/homebridge-appletv-enhanced/blob/f0395a352571ded4c893ef3d2a15d3453379a261/src/appleTVEnhancedAccessory.ts#L1094

### Expected Behavior

I'd expect the plugin to wait for the backoff time before retrying

### Steps To Reproduce

I have just installed the plugin in tried to pair my Apple TV

### Logs

```Markdown
[17/12/2023, 12:01:11] [Apple TV Enhanced] *** (****): stdout: 2023-12-17 12:01:11 ERROR [pyatv.scripts.atvremote]: Pairing failed
Traceback (most recent call last):
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/support/__init__.py", line 71, in error_handler
    return await func(*args, **kwargs)
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/protocols/companion/auth.py", line 63, in start_pairing
    pairing_data = _get_pairing_data(resp)
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/protocols/companion/auth.py", line 33, in _get_pairing_data
    raise exceptions.AuthenticationError(stringify(tlv))
pyatv.exceptions.AuthenticationError: Error=BackOff, BackOff=1548s, SeqNo=M2

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/scripts/atvremote.py", line 196, in pair
    await self._perform_pairing(pairing)
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/scripts/atvremote.py", line 206, in _perform_pairing
    await pairing.begin()
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/protocols/companion/pairing.py", line 47, in begin
    await error_handler(
  File "/var/lib/homebridge/appletv-enhanced/.venv/lib/python3.9/site-packages/pyatv/support/__init__.py", line 77, in error_handler
    raise fallback(str(ex)) from ex
pyatv.exceptions.PairingError: Error=BackOff, BackOff=1548s, SeqNo=M2
```

### Configuration

```JSON
{
            "mediaTypes": [
                "music",
                "video"
            ],
            "deviceStates": [
                "paused",
                "playing"
            ],
            "remoteKeysAsSwitch": [
                "turn_off",
                "turn_on"
            ],
            "customInputURIs": [],
            "discover": {
                "unicast": [],
                "blacklist": [],
                "multicast": true
            },
            "name": "Apple TV Enhanced",
            "avadaKedavraAppAmount": 15,
            "disableVolumeControlRemote": false,
            "forceVenvRecreate": false,
            "logLevel": 3,
            "platform": "AppleTVEnhanced"
        }
```

### Environment

- OS: raspberry 2
- Homebridge: 1.7.0
- Apple TV Enhanced: 0.7.0	
- Node: 20.10.0
- npm: 10.2.3
- Python: 3.9.2

Trying to pair an Apple TV 4K (gen 3)

**Comment by courentin** [comment]
For additional context, I tried to turn off the plugin to wait for the backoff time and re-enabling it back, I now have the following error at first attempt: `pyatv.exceptions.PairingError: Error=Authentication, SeqNo=M4`

and then back looping to the `pyatv.exceptions.AuthenticationError: Error=BackOff, BackOff=1548s, SeqNo=M2` errors

**Comment by maxileith** [comment]
Please post the complete debug logs from the start of the plugin until the error occurs.

**Comment by maxileith** [comment]
I have just recreated the bug, you were right. The problem should be fixed in v0.7.1-1. Could you please verify that?

**Comment by courentin** [comment]
Hi @maxileith yes I can confirm that it is working on 0.7.1-1
Thanks a lot!!
